src/Compilers/CSharp/Portable/Binder/Semantics/AccessCheck.cs

```diff
@@ -221,6 +221,11 @@ private static bool IsNamedTypeAccessible(NamedTypeSymbol type, Symbol within, r
 
             failedThroughTypeCheck = false;
 
+            if (containingType.IsTupleType)
+            {
+                containingType = containingType.TupleUnderlyingType;
```

Comment:
Can this leak the underlying type into error messages?

Perhaps we can assume/require that ValueTuple is a public type and then not do this unwrapping to underlying?